app/javascript/pages/Affiliates/New.tsx

```diff
@@ -0,0 +1,276 @@
+import { router, usePage } from "@inertiajs/react";
+import cx from "classnames";
+import * as React from "react";
+
+import { addAffiliate, SelfServeAffiliateProduct } from "$app/data/affiliates";
+import { asyncVoid } from "$app/utils/promise";
+import { assertResponseError } from "$app/utils/request";
+import { isValidEmail } from "$app/utils/email";
+import { isUrlValid } from "$app/utils/url";
+
+import { Button } from "$app/components/Button";
+import { Icon } from "$app/components/Icons";
+import { NavigationButtonInertia } from "$app/components/NavigationButton";
+import { useLoggedInUser } from "$app/components/LoggedInUser";
+import { NumberInput } from "$app/components/NumberInput";
+import { showAlert } from "$app/components/server-components/Alert";
+import { PageHeader } from "$app/components/ui/PageHeader";
+import { Table, TableBody, TableCell, TableHead, TableHeader, TableRow } from "$app/components/ui/Table";
+
+type AffiliateProduct = SelfServeAffiliateProduct & { referral_url: string };
+
+type Props = {
+  products: AffiliateProduct[];
+  affiliates_disabled_reason: string | null;
+};
+
+export default function AffiliatesNew() {
+  const props = usePage<{ props: Props }>().props as unknown as Props;
+  const loggedInUser = useLoggedInUser();
+  const [isSubmitting, setIsSubmitting] = React.useState(false);
+  const [email, setEmail] = React.useState("");
+  const [products, setProducts] = React.useState<AffiliateProduct[]>(props.products);
+  const [applyToAllProducts, setApplyToAllProducts] = React.useState(props.products.length === 0);
```

Comment:
There was a global `affiliate` state before. Can you explain in your self review the reasons behind this decision to split it to dedicated state for `products` and `applyToAllProducts` ?